### PR TITLE
 * Use suggested Antiforgery AJAX patern.

### LINKED ISSUE #580: Update use of Antiforgery + Ajax to built-in header support
Our use of antiforgery for ajax in this sample is a bit ad-hoc. https://github.com/aspnet/MusicStore/blob/5a3c4957d9707a9901891a6b909863d1a34cc37f/src/MusicStore/Controllers/ShoppingCartController.cs#L72

We have built in support for using a header for antiforgery in rc2, we should update the sample to use our recommended pattern.

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #580

### CODE REVIEW

File: src/MusicStore/Views/ShoppingCart/Index.cshtml [hunk 1]
```diff
@@ -6,10 +6,10 @@
 
 @functions
 {
-    public string GetAntiXsrfToken()
+    public string GetAntiXsrfRequestToken()
     {
         var tokens = Xsrf.GetTokens(Context);
```
Comment (rynowak) [inline src/MusicStore/Views/ShoppingCart/Index.cshtml]: What does this do?

File: src/MusicStore/Startup.cs [hunk 2]
```diff
@@ -144,11 +146,19 @@ public void ConfigureProduction(IApplicationBuilder app, ILoggerFactory loggerFa
...
-        public void Configure(IApplicationBuilder app)
+        public void Configure(IApplicationBuilder app, IAntiforgery antiforgery)
         {
+            app.Use(next => context =>
+            {
+                var requestToken = antiforgery.GetAndStoreTokens(context).RequestToken;
+                context.Response.Cookies.Append("XSRF-TOKEN", requestToken, new CookieOptions() { HttpOnly = false });
+
+                return next(context);
+            });
```
Comment (rynowak) [inline src/MusicStore/Startup.cs]: Why is this setting the token for every request?

**Comment by ryanbrandenburg** [comment]
cc @rynowak

File: src/MusicStore/Views/ShoppingCart/Index.cshtml (hunk 1 above)
Comment (ryanbrandenburg) [inline src/MusicStore/Views/ShoppingCart/Index.cshtml]: This is getting the XSRF token for use in AJAX below. Are you saying there's a better way to get it on the page or that we can inline it down there?

**Comment by ryanbrandenburg** [comment]
:up: :date: with what we talked about @rynowak. Also I guess technically I should cc @blowdart just in case I'm going something that's not best practice.

File: src/MusicStore/Startup.cs [hunk 3]
```diff
@@ -144,10 +145,10 @@ public void ConfigureProduction(IApplicationBuilder app, ILoggerFactory loggerFa
 
             app.UseExceptionHandler("/Home/Error");
 
-            Configure(app);
+            Configure(app, antiforgery);
         }
 
-        public void Configure(IApplicationBuilder app)
```
Comment (rynowak) [inline src/MusicStore/Startup.cs]: Revert changes here

**Comment by rynowak** [comment]
:shipit:

**Pull request merged**